odoo/tools/mail.py

```diff
@@ -184,6 +185,9 @@ def _tag_matching_regex_in_text(regex, node, tag='span', attrs=None):
             idx = item.end()
             node_idx = node_idx + 1
 
+    if el.get('data-o-mail-quote') or el.get('data-o-mail-quote-node'):
+        return
+
```

Comment:
I believe this fixes the issue because the user signature is pre-quoted (because it's a sanitized field itself).

This prevents the outer div of the message from being marked a "quote container", which is good. But won't it also prevent it from being marked appropriately in legitimate cases where we do want to quote the rest of the container? To me it looked like the intent was that the text in the same block as the signature should be quoted too in case you want to add some stuff after your signature.

Not a game breaker but it's *a* change.

I'm just not really convinced that the approach is logically sound but I'm also biased from working on another approach so idk
This one is certainly cheaper.
This one is certainly a bit cheaper

Reply:
It doesn't do a change the second time so long as you don't allow it to run apply the "data-mail-quote" attribute to elements that came before the original "quote" element. That's what I'm avoiding in my fix and what my test is trying to test against.

The original logic isn't really sound, ideally we should rewrite it to avoid any kind of "modifying the structure we're iterating on" business. It should just look a pattern, then mark all nodes as quotes at once without conditional recursivity.

But that's the state of things now, so I think just making it behave as it did before in stable is enough.